### ISSUE TITLE
Update 9.X - Bug

### ISSUE THREAD

**Issue by marcottedan** [PhilippeKr/TorchLight#41]
Module now throws this error when you light a torch:

```
torchlight.js:380 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'type')
[Detected 1 package: torchlight]
    at storeTokenLighting (torchlight.js:380)
    at onButtonClick (torchlight.js:205)
```

**Comment by coffiarts** [comment]
Same for me. None of the lights switch on or off in v9.

**Comment by PhluXx1** [comment]
@marcottedan @coffiarts Guys I hate to tell you but this project is effectively abandoned. The only reason it even received a 0.8.x update was due to a pull request. Unless someone else provides a v9 update I wouldn't expect to see one.

**Comment by coffiarts** [comment]
Good to know, thanks for the update anyway. I appreciate your efforts. Still, this add-on is/was a really cute thing! I'd be delighted to see some hero pick it up again :)

**Comment by PhluXx1** [comment]
@coffiarts I'm not the dev just a fan and yeah it sucks because I love the module hopefully someone takes it over.

**Comment by Malekal4699** [comment]
Alright, I did not make a Pull request as the last 2 were not processed. So I forked and updated to v9.

You can grab it here: 

https://github.com/Malekal4699/TorchLight/releases/latest/download/module.json

Have fun :)

**Comment by PhluXx1** [comment]
> Alright, I did not make a Pull request as the last 2 were not processed. So I forked and updated to v9.
> 
> You can grab it here:
> 
> https://github.com/Malekal4699/TorchLight/releases/latest/download/module.json
> 
> Have fun :)

Thanks bro you're the real MVP.

@Malekal4699 May not be a bad idea to for you to make a post about this update on the Foundry [subreddit](https://www.reddit.com/r/FoundryVTT/) to get the word out to people.

**Comment by coffiarts** [comment]
@Malekal4699 - Now YOU are our hero!
Checked your version out, and from a brief test it seems to work like charm!
Took me a bit of a brain to figure out how to install your update over PhillippeKr's original, as your directory structure differs from his. Took me a couple of minutes to understand that I need to completely replace any files under ...\data\mdules\torchlight with yours.
Thanks a thousand times!!

**Comment by Malekal4699** [comment]
> @Malekal4699 - Now YOU are our hero!
> 
> Checked your version out, and from a brief test it seems to work like charm!
> 
> Took me a bit of a brain to figure out how to install your update over PhillippeKr's original, as your directory structure differs from his. Took me a couple of minutes to understand that I need to completely replace any files under ...\data\mdules\torchlight with yours.
> 
> Thanks a thousand times!!

It's usually a good idea, when having issue after an update, to simply uninstall the module and reinstall with the new manifest url

**Comment by coffiarts** [comment]
> It's usually a good idea, when having issue after an update, to simply uninstall the module and reinstall with the new manifest url

But this is the thing I didn't get (probably out of my stupidity): I can't find a valid manifest url on your github page at https://github.com/Malekal4699/TorchLight. That's why I had to fumble around manually.
Did I just miss it?

**Comment by PhluXx1** [comment]
> https://github.com/Malekal4699/TorchLight/releases/latest/download/module.json

@coffiarts You could just take the URL he provided above and just paste it into the Manifest URL option at the bottom of the Install Module installer in Foundry then click install (see screenshot below).

![image](https://user-images.githubusercontent.com/72770484/147672607-0389e81a-cb1d-482c-aded-7bcf2a977d4b.png)

**Comment by Malekal4699** [comment]
Like @PhluXx1 said. I'll pick the module up, so the Foundry should soon point at the right place.

**Comment by PhluXx1** [comment]
> I'll pick the module up, so the Foundry should soon point at the right place.

@Malekal4699 The news just gets better and better. If I could recommend a name change to "LampLight" or something like that to differentiate and to keep with the tradition of take overs. Thank you for your service.

![salutegif](https://user-images.githubusercontent.com/72770484/147684058-fe57564b-8915-4434-9078-136f685263a6.gif)

**Comment by Malekal4699** [comment]
HeaUDLight?

**Comment by PhluXx1** [comment]
> HeaUDLight

You mind elaborating on this name?

**Comment by Malekal4699** [comment]
Headlight, Plus HUD, Head's up display which is the main functionality of the mod lol

**Comment by Malekal4699** [comment]
HeUaDLight

**Comment by PhluXx1** [comment]
> Headlight, Plus HUD, Head's up display which is the main functionality of the mod lol
> HeUaDLight

@Malekal4699 Name it whatever you're happiest with, personally I think HUDLight sounds better

**Comment by coffiarts** [comment]
> https://github.com/Malekal4699/TorchLight/releases/latest/download/module.json

Yeah, that worked. Somehow I was sitting on my eyes. Thanks again! :)

**Comment by coffiarts** [comment]
> > https://github.com/Malekal4699/TorchLight/releases/latest/download/module.json
> 
> @coffiarts You could just take the URL he provided above and just paste it into the Manifest URL option at the bottom of the Install Module installer in Foundry then click install (see screenshot below).
> 
> ![image](https://user-images.githubusercontent.com/72770484/147672607-0389e81a-cb1d-482c-aded-7bcf2a977d4b.png)

@PhluXx1 -  Yeah, that worked. Somehow I was sitting on my eyes. Thanks again! :)

**Comment by dczwardCZAR** [comment]
OMG, thank you for fixing this for v9. I've been wanting to use this for a long time ever since in broke back in v8.? . But this would be fantastic if it could consume a torch inventory item.

**Comment by Malekal4699** [comment]
> OMG, thank you for fixing this for v9. I've been wanting to use this for a long time ever since in broke back in v8.? . But this would be fantastic if it could consume a torch inventory item.

Hi, please add an issue in the new repo so I don't forget! 

Thanks!

**Comment by jpmiers** [comment]
> Alright, I did not make a Pull request as the last 2 were not processed. So I forked and updated to v9.
> 
> You can grab it here:
> 
> https://github.com/Malekal4699/TorchLight/releases/latest/download/module.json
> 
> Have fun :)

Hello, I've installed this fork and it seems that the mod is enabling debug mode in the console. Is there any way to turn it off? It's spitting out messages anytime I click or move the mouse. Lots of clutter. I verified that its this mod by disabling it and reenabling it.

**Comment by PhluXx1** [comment]
@jpmiers Open up a new issue on the github page https://github.com/Malekal4699/LightsHUD/
